### ISSUE TITLE
Pruned TFLite model has invalid model identifier after post training quantisation

### ISSUE THREAD

**Issue by niciBume** [tensorflow/tensorflow#52263]
### 1. System information

- Linux Ubuntu 20.04
- Pip TensorFlow package version 2.6.0

### 2. Code
Reproducable code in this [gist](https://colab.research.google.com/gist/niciBume/2fb8d9eb36d5e13d8faaea9e1273e980/tensorflow-lite-debugger-colab.ipynb#scrollTo=RD0CEfccGN2C)

### 3. Description
After pruning a keras model according to [tf colab](https://www.tensorflow.org/model_optimization/guide/pruning/pruning_with_keras), the post training quantized tflite can't be invoked (the quantisation works for the un-pruned model).

`interpreter` = tf.lite.Interpreter(model_path=MODEL_TF_PRUNE)`

Leads to the error:
```
ValueError                                Traceback (most recent call last)

<ipython-input-56-edee2625144a> in <module>()
----> 1 interpreter = tf.lite.Interpreter(model_path=MODEL_TF_PRUNE)

/usr/local/lib/python3.7/dist-packages/tensorflow/lite/python/interpreter.py in __init__(self, model_path, model_content, experimental_delegates, num_threads, experimental_op_resolver_type, experimental_preserve_all_tensors)
    365           _interpreter_wrapper.CreateWrapperFromFile(
    366               model_path, op_resolver_id, custom_op_registerers_by_name,
--> 367               custom_op_registerers_by_func, experimental_preserve_all_tensors))
    368       if not self._interpreter:
    369         raise ValueError('Failed to open {}'.format(model_path))

ValueError: Model provided has model identifier '

', should be 'TFL3'
```

**Comment by Saduf2019** [comment]
@niciBume 
Please refer to these solutions and let us know: [link](https://github.com/tensorflow/tensorflow/issues/32204#issuecomment-703506041),[link1](https://stackoverflow.com/questions/56857400/converting-mobilenet-segmentation-model-to-tflite)

**Comment by niciBume** [comment]
Oh wow guys... I found the error. 
Problem was on [this line](https://colab.research.google.com/gist/niciBume/2fb8d9eb36d5e13d8faaea9e1273e980/tensorflow-lite-debugger-colab.ipynb#scrollTo=YT8OIvQiKDhg&line=1&uniqifier=1) I was feeding the interpreter a `.h5` model and not the actual pruned TFlite file.

Sorry for wasting your time!

TLDR: Quantizing pruned networks works if you feed the interpreter the correct model...
